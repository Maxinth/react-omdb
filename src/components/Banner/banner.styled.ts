import styled from "styled-components";

const BannerText = styled.h1``;
const Container = styled.div`
  width: 100%;
  min-height: 100%;
  object-fit: cover;
  /* width: 320px; */
  display: flex;
  justify-content: center;
  align-items: center;
  height: 257px;
  background-repeat: no-repeat;
  /* background-attachment: fixed; */
  background-image: url("../../../Rectangle-5.svg");
  /* background-position: 50% 50%; */
  ${BannerText} {
    color: #ffffff;
    text-align: center;
    font-family: "DM Sans", sans-serif;
    font-size: 28px;
    line-height: 36.46px;
    font-weight: bold;

    @media (min-width: 321px) {
      font-size: 72px;
      line-height: 93.74px;
    }

    @media (min-width: 1440px) {
      text-align: left;
      width: 490px;
      height: 282px;
    }
  }

  @media (min-width: 321px) {
    background-image: url("../../../Rectangle-5-2.png");
    padding: 109px 172px 159px;
    object-fit: contain;
    height: 550px;
  }

  @media (min-width: 835px) {
    object-fit: contain;
    /* background-position: 50% 50%; */
    background-image: url("../../../Rectangle-5.png");
  }

  @media (min-width: 1440px) {
    justify-content: flex-start;
    padding: 109px 873px 159px 77px;
  }
`;
export { BannerText, Container };
