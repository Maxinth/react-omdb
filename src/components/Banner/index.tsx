import { Container, BannerText } from "./banner.styled";
import React from "react";
const Banner = () => {
  return (
    <Container>
      <BannerText>Watch something incredible</BannerText>
    </Container>
  );
};

export default Banner;
