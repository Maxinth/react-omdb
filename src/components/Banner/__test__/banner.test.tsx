import renderer from "react-test-renderer";
import Banner from "..";
import React from "react";

it("renders a snapshot of <Banner />", () => {
  const banner = renderer.create(<Banner />).toJSON();
  expect(banner).toMatchSnapshot();
});
