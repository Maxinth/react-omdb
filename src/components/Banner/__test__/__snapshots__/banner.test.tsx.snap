// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <Banner /> 1`] = `
<div
  className="sc-gsDKAQ bYfuev"
>
  <h1
    className="sc-bdvvtL"
  >
    Watch something incredible
  </h1>
</div>
`;
